src/Middleware/RequestQueue/src/SemaphoreWrapper.cs

```diff
@@ -0,0 +1,33 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.RequestQueue
+{
+    public class SemaphoreWrapper
```

Comment:
```suggestion
    internal class SemaphoreWrapper
```

Let's make this internal and use `InternalsVisibleTo` (@halter73 can help) since this kind of type wouldn't be a public type in the final product.